### PR TITLE
Updated dependencies

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes part of #2106 
| Related issues/PRs | https://github.com/doctrine/DoctrinePHPCRBundle/pull/267
| License | MIT
| Documentation PR | none

#### What's in this PR?

This PR updates dependencies and fixes last deprecations in sulu.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -17,19 +17,20 @@
...
         "guzzlehttp/guzzle": "^6.2",
         "imagine/imagine": "~0.6.1",
         "massive/search-bundle": "0.16.*",
         "sulu/document-manager": "dev-develop",
         "sensio/framework-extra-bundle": "3.0.*",
-        "stof/doctrine-extensions-bundle": "1.1.*",
+        "stof/doctrine-extensions-bundle": "~1.1",
         "symfony-cmf/routing-bundle": "1.2.*",
         "symfony/symfony": "~2.8.7, !=2.8.10",
+        "symfony/security-acl": "~2.8",
```
Comment (danrot) [inline composer.json]: What do we need that for?

File: composer.json [hunk 2]
```diff
@@ -44,22 +45,22 @@
...
         "guzzle/plugin-mock": "3.9.*",
-        "massive/build-bundle": "0.2.*",
+        "massive/build-bundle": "dev-master",
         "matthiasnoback/symfony-config-test": "^1.0",
         "matthiasnoback/symfony-dependency-injection-test": "^0.7.3",
         "phpunit/phpunit": ">=4.8.0",
         "symfony/phpunit-bridge": "~2.8",
         "sensio/framework-extra-bundle": "~3.0",
-        "symfony/monolog-bundle": "2.4.*",
+        "symfony/monolog-bundle": "~2.4",
```
Comment (danrot) [inline composer.json]: Shouldn't that have the same value as in the `suggest` section?

File: src/Sulu/Bundle/ContentBundle/Tests/Unit/Content/InternalLinksContainerTest.php [hunk 3]
```diff
@@ -42,14 +42,7 @@ protected function setUp()
...
         $this->container = new InternalLinksContainer(
-            [2, 3, 1],
-            $this->executor->reveal(),
-            $this->builder->reveal(),
-            [],
-            new NullLogger(),
-            'default',
-            'en',
-            true
+            [2, 3, 1], $this->executor->reveal(), $this->builder->reveal(), [], new NullLogger(), 'default', 'en', true
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Tests/Unit/Content/InternalLinksContainerTest.php]: I'd say the previous code was correct from a code style perspective.

File: src/Sulu/Bundle/ContentBundle/Tests/Unit/Content/InternalLinksContainerTest.php [hunk 4]
```diff
@@ -63,14 +56,7 @@ public function testGetDataDraftAndPublished()
...
         $this->container = new InternalLinksContainer(
-            [2, 3, 1],
-            $this->executor->reveal(),
-            $this->builder->reveal(),
-            [],
-            new NullLogger(),
-            'default',
-            'en',
-            false
+            [2, 3, 1], $this->executor->reveal(), $this->builder->reveal(), [], new NullLogger(), 'default', 'en', false
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Tests/Unit/Content/InternalLinksContainerTest.php]: Same as above

File: src/Sulu/Bundle/ContentBundle/Tests/Unit/Content/InternalLinksContainerTest.php [hunk 5]
```diff
@@ -88,14 +74,7 @@ public function testGetDataOrder()
...
```
Comment (danrot) [inline src/Sulu/Bundle/ContentBundle/Tests/Unit/Content/InternalLinksContainerTest.php]: Same as above

File: src/Sulu/Bundle/WebsiteBundle/Controller/DefaultController.php [hunk 6]
```diff
@@ -41,44 +40,4 @@ public function indexAction(StructureInterface $structure, $preview = false, $pa
...
     }
-
-    /**
-     * Creates a redirect for configured webspaces.
-     *
-     * @param Request $request
-     *
-     * @return Response
-     *
-     * @deprecated since 1.2 use SuluWebsiteBundle:Redirect:redirectWebspace instead
```
Comment (danrot) [inline src/Sulu/Bundle/WebsiteBundle/Controller/DefaultController.php]: Would you already delete this? Would consider this a BC break, and would not do that before 2.0. Are any of these methods containing deprecated Symfony code?

Comment (wachterjohannes) [inline src/Sulu/Bundle/WebsiteBundle/Controller/DefaultController.php]: yes both of them (: the `$this->forward` will do something different which is quite difficult to solve in unit tests without "version" flag. I thought about that we should use this version to "collect" smaller upgrades which doesn't hurt which we has already done in develop.

See:
* https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php#L64
* https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php#L68

File: composer.json (hunk 1 above)
Comment (wachterjohannes) [inline composer.json]: we need that for the update of symfony3 where this component is a own dependency. but we should not introduce it until we have symfony3 as dependency. i will remove it.

Comment (danrot) [inline composer.json]: But the point is that we don't use the ACL at all, so it should not be here in any case.

File: .travis.yml [hunk 7]
```diff
@@ -53,9 +56,9 @@ before_script:
     fi
   # the content tests are intensive and there are memory leaks, this is more pronounced with the Jackalope DBAL PHPCR implementation.
   - echo "memory_limit=2048M" >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
-  - phpenv config-rm xdebug.ini
   - composer self-update
-  - composer install  --prefer-dist --no-interaction
+  - composer update $COMPOSER_FLAGS
+  - if [[ $SYMFONY__PHPCR__TRANSPORT = jackrabbit ]]; then composer require jackalope/jackalope-jackrabbit:~1.2 ; fi
```
Comment (wachterjohannes) [inline .travis.yml]: add it to "composer.json" before `composer update`

**Comment by wachterjohannes** [comment]
@danrot this can now be reviewed.

**Comment by danrot** [comment]
@wachterjohannes Can you tick of all the fixed dependencies on #2106? Thank you.

**Pull request merged**